number/src/lib.rs

```diff
@@ -31,6 +33,52 @@ pub fn log2_exact(n: BigUint) -> Option<usize> {
         .filter(|zeros| n == (BigUint::from(1u32) << zeros))
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct VariablySizedColumn<F> {
+    column_by_size: BTreeMap<usize, Vec<F>>,
+}
+
+#[derive(Debug)]
+pub struct HasMultipleSizesError;
+
+impl<F> VariablySizedColumn<F> {
+    /// Create a view where each column has a single size. Fails if any column has multiple sizes.
+    pub fn get_only_size(&self) -> Result<&Vec<F>, HasMultipleSizesError> {
+        if self.column_by_size.len() != 1 {
+            return Err(HasMultipleSizesError);
+        }
+        Ok(self.column_by_size.values().next().unwrap())
+    }
+}
+
+pub fn get_only_size<F>(
```

Comment:
Maybe `get_uniquely_sized`? With this name, it could mean that it only returns the size.

Reply:
Done